test/queries/gossip_node_query_test.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class GossipNodeQueryTest < ActiveSupport::TestCase
+  setup do
+    @network = "mainnet"
+
+    @node = create(
+      :gossip_node,
+      network: @network,
+      staked: true,
+      identity: "exampleIdentity"
+    )
+    create(:gossip_node, network: "testnet")
+    create(:gossip_node, network: "mainnet", staked: false)
+
+    @query_keys = [
```

Comment:
I'd add `attr_reader` or constant or class variable in GossipNodeQuery and use it here, because now you need to update two places.